pkg/planner/core/find_best_task.go

```diff
@@ -186,7 +187,14 @@ func enumeratePhysicalPlans4Task(
 	if _, ok := p.Self().(*logicalop.LogicalSequence); ok {
 		iteration = iterateChildPlan4LogicalSequence
 	}
-
+	var fd *funcdep.FDSet
+	if addEnforcer {
+		switch logicalPlan := p.Self().(type) {
+		case *logicalop.LogicalJoin, *logicalop.LogicalAggregation:
```

Comment:
If you want to control the effect scope of this enhance, it is better to only support the LogicalAggregation in This PR.

Reply:
We can take a look at the effect first and make modifications before the official release.